packages/liveblocks-client/src/room.ts

```diff
@@ -792,6 +797,7 @@ export function makeStateMachine<
       return null;
     }
 
+    // auth prepared here, a Promise, but not resolved yet.
```

Comment:
Technically, it returns a function that returns a Promise. We "create" a promise only when `auth` is called.